clarity_ext/domain/aliquot.py

```diff
@@ -46,13 +46,22 @@ def create_well_from_rest(resource, container_repo):
 
 class Sample(DomainObjectMixin):
 
-    def __init__(self, sample_id):
+    def __init__(self, sample_id, name, project):
         self.id = sample_id
+        self.name = name
+        self.project = project
 
     def __repr__(self):
         return "<Sample id={}>".format(self.id)
 
     @staticmethod
     def create_from_rest_resource(resource):
-        sample = Sample(resource.id)
+        project = Project(resource.project.name) if resource.project else None
```

Comment:
So the design is (on a high level) this:

- Clarity provides an API, from which we get the API resources. These are simple data contract objects. We use the Scilifelab/Stockholm library (genologics) for accessing those
- We got domain classes that may be "thick" business objects, which can be complex and contain references to other objects without issues. These will be used by the end user configuring the system.
- We get `repository` classes which fetch data from the API and map them to domain classes. The mapping happens through factory methods like the one we're looking at now.
- We got `services` which fetch data from the repositories. In general, unit testing them by faking only the repositories should be enough to test most of the business logic.